.github/workflows/ci.yml

```diff
@@ -0,0 +1,285 @@
+name: Tests master
+
+on:
+  # This avoids having duplicate builds for a pull request
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  ############################################################################
+  # Lint jobs
+  ############################################################################
+  lint-py:
+    name: Lint Python
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false 
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
+      - name: Cache pip
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-py-3.8-pip-${{ hashFiles('**/setup.py') }}
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.8'
+          architecture: 'x64'
+      - name: Install python dependencies
+        run: python -m pip install --editable .[test]
+      - name: Show python environment
+        run: |
+          python --version
+          python -m pip list
+      - name: Run python lint
+        run: pylint lektor
+
+  lint-js:
+    name: Lint JS
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
+      - name: Cache node
+        uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-13-${{ hashFiles('**/lektor/admin/package-lock.json') }}
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '13'
+      - name: Install node dependencies
+        run: |
+          cd lektor/admin/
+          npm install .
+          npm run webpack
+      - name: Show node environment
+        run: |
+          node --version
+          npm --version
+      - name: Run JS lint
+        run: |
+          cd lektor/admin/
+          npm run lint
+
+  ############################################################################
+  # Node tests
+  ############################################################################
+  linux-node:
+    name: Linux node${{ matrix.NODEJS_VERSION }} tests
+    runs-on: ubuntu-latest
+    needs: lint-js
+    env:
+      CI: True
+      NODEJS_VERSION: ${{ matrix.NODEJS_VERSION }}
+    strategy:
+      fail-fast: false 
+      matrix:
+        NODEJS_VERSION: ['10', '12', '13']
```

Comment:
It runs in parallel containers so it makes no difference in time

Reply:
But there's really no point to running them more than once. It still wastes resources and introduces a mental overhead for any contributor who see 20 checks of which 9 are the same thing.

Reply:
I think there is a point to run them all. Since all are still supported we may have contributors who use them. There are differences between them and sometimes one of them may break. I really think we should test for all supported releases here.

I do however think it's OK to only test latest for macOS and Windows and all on Linux. That way we reduce the amount of build jobs but can still catch version inconsistencies.

Though I don't really mind keeping all the tests either.

Reply:
> I think there is a point to run them all. Since all are still supported we may have contributors who use them. There are differences between them and sometimes one of them may break.

The way I see it, what we're testing with these runs is whether Webpack and Babel work on all these versions of Node. Which seems a bit overkill for me, since those projects already test for that. But if these builds turn out to be non-flaky there's not much harm either :)

Reply:
I see what you mean. What this protect us from is if someone makes a PR that updates a dependency that breaks some Node version we still want to support. JS projects are infamous for breaking backwards compatibility so I think we should still consider this.

In the future I think we can be a bit more clever and only run JS tests if the commit touches JS files and vice versa.

Reply:
Agreed, we need JS tests in the first place :-p